packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/OpenShiftVirtualMachinesList.tsx

```diff
@@ -34,6 +35,19 @@ const openShiftVmFieldsMetadataFactory: ResourceFieldFactory = (t) => [
     },
     sortable: true,
   },
+  {
+    resourceFieldId: 'template',
+    jsonPath: (data: VmData) =>
+      (data?.vm as OpenshiftVM)?.object?.metadata?.labels?.['vm.kubevirt.io/template'] ?? '',
```

Comment:
a. we should not need to cast here
b. this is actually just a json path, why do you use a method ? what am i missing ?

Reply:
try:
 ```js
 jsonPath: ['object', 'metadata' ,'labels', 'vm.kubevirt.io/template'],
```

Reply:
> a. we should not need to cast here
>

The proposed solution with generics was dropped. If you have any other idea please share.

> b. this is actually just a json path, why do you use a method ? what am i missing ?
>

To benefit from the type system and avoid i.e. typos.

Reply:
the idea of json path is to make the code clear and clean, if you add a method here, you add casting, you add a long `.?` code snippet, I prefer using jsonPath if possible.

Reply:
> the idea of json path is to make the code clear and clean
>

IMHO the opposite is true i.e. as stated above we lose type safety here. 

>  if you add a method here, you add casting, you add a long .? code snippet
>

1. Casting is needed because we dropped generics in `VMCellProps` that were implemented in the VM list screens. Compare the code above with that version:
  https://github.com/rszwajko/forklift-console-plugin/blob/fb778cf9e4b2ec24be6c754122233cda013de594/packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/VSphereVirtualMachinesRow.tsx#L39
2. Long path is needed because we dropped "flat" object approach.  Using a string(jsonpath) is here similar to standard JS syntax.

> I prefer using jsonPath if possible.
>

OK. Note that currently array of strings is not accepted - it's either a string or a method.